spec/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement-spec.js

```diff
@@ -32,6 +44,30 @@ describe('Youtube link enhancement', function () {
     expect($element.find($toReplace).length).toBe(2)
   })
 
+  it('doesn\'t replace links when campaign cookies are turned off', function () {
+    window.GOVUK.cookie('cookie_policy', '{ "campaigns": false }')
+    var $element = $('<div><p><a href="https://www.youtube.com/watch?v=0XpAtr24uUQ">Agile at GDS</a></p></div>')
+    var $toReplace = $element.find('p, a')
+
+    var enhancement = new GOVUK.GovspeakYoutubeLinkEnhancement($element)
+    enhancement.init()
+
+    expect($element.find('.media-player .video').length).toBe(0)
+    expect($element.find($toReplace).length).toBe(2)
+  })
+
+  it('replaces a link and it\'s container with a media-player embed when campaign cookies are turned on', function () {
+    window.GOVUK.cookie('cookie_policy', '{ "campaigns": true }')
+    var $element = $('<div><p><a href="https://www.youtube.com/watch?v=0XpAtr24uUQ">Agile at GDS</a></p></div>')
+    var $toReplace = $element.find('p, a')
```

Comment:
Is it worth DRYing these two lines up into the beforeEach?

Reply:
not sure, as they're not applicable to all the test (like the one not embedding non YT videos)